File: app/src/main/java/com/keplersegg/myself/Room/Dao/EntryDao.java
package com.keplersegg.myself.Room.Dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.keplersegg.myself.Room.Entity.Entry;

import java.util.List;

@Dao
public interface EntryDao {

    @Query("SELECT * FROM Entry")
    List<Entry> getAll();

    @Query("SELECT * FROM Entry where Day = :day and TaskId = :taskId")
    Entry get(int day, int taskId);

    @Query("SELECT COUNT(0) from Entry")
    int getCount();

    @Insert
    void insertAll(Entry... entries);

    @Delete
    void delete(Entry entry);
}

File: app/src/main/java/com/keplersegg/myself/Room/Dao/EntryDao.kt
package com.keplersegg.myself.Room.Dao

import com.keplersegg.myself.Room.Entity.Entry

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface EntryDao {

    @get:Query("SELECT * FROM Entry")
    val all: List<Entry>

    @get:Query("SELECT COUNT(0) from Entry")
    val count: Int

    @Query("SELECT * FROM Entry where Day = :day and TaskId = :taskId")
    operator fun get(day: Int, taskId: Int): Entry?

    @Insert
    fun insert(entry: Entry)

    @Update
    fun update(entry: Entry)

    @Delete
    fun delete(entry: Entry)

    @Query("SELECT TaskId, sum(Value) AS EntryCount FROM Entry group by TaskId")
    fun getCounts(): List<TaskEntryCount>

    class TaskEntryCount {
        var TaskId: Int? = null
        var EntryCount: Int? = null
    }
}
